src/librbd/internal.cc

```diff
@@ -679,11 +679,26 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
       for (auto &id_it : info.second) {
 	string name;
 	r = cls_client::dir_get_name(&ioctx, RBD_DIRECTORY, id_it, &name);
-	if (r < 0) {
-	  lderr(cct) << "Error looking up name for image id " << id_it
-		     << " in pool " << info.first.second << dendl;
-	  return r;
+	if (r < 0 && r != -ENOENT) {
+          lderr(cct) << "Error looking up name for image id " << id_it
+                     << " in pool " << info.first.second << dendl;
+          return r;
+        } else if (r == -ENOENT) {
+          cls::rbd::TrashImageSpec trash_spec;
+          r = cls_client::trash_get(&ioctx, id_it, &trash_spec);
+          if (r < 0) {
+            if (r != -EOPNOTSUPP && r != -ENOENT) {
+              lderr(cct) << "Error looking up name for image id " << id_it
+                         << " in rbd trash" << dendl;
+              return r;
+            }
+            return -ENOENT;
+          }
+
+          // the child image is in trash bin
+          continue;
```

Comment:
I don't think we want to hide the fact that the image is in the trash -- but if the image is in the trash you really need to show the image id in the "rbd children" output since you could technically have lots of images with the same name in the trash. Therefore, I think it would be nice to add new ```rbd_list_children2``` and ```librbd::RBD::list_children2``` methods that return a structure containing the pool id, image name, and image id. Then, the rbd CLI could use the new API method combined w/ the forthcoming "trash_get" API to determine if the child is in the trash.